php/commands/src/CLI_Command.php

```diff
@@ -575,4 +575,96 @@ private function get_update_type_str( $assoc_args ) {
 		return $update_type;
 	}
 
+	/**
+	 * Detects if a command exists
+	 *
+	 * This commands checks if a command is registered with WP-CLI.
+	 * If the command is found then it returns with exit status 0.
+	 * If the command doesn't exist, then it will exit with status 1.
+	 *
+	 * ## OPTIONS
+	 * <command_name>...
+	 * : The command
+	 *
+	 * ## EXAMPLES
+	 *     wp cli has-command "wp site delete"
+	 *     wp cli has-command wp site delete
+	 *
+	 * @subcommand has-command
+	 *
+	 * @when before_wp_load
+	 *
+	 * @param array $_          Array of positional arguments.
+	 * @param array $assoc_args Array of associative arguments.
+	 */
+	public function has_command( $_, $assoc_args ) {
+
+		/**
+		 * Get dump as array.
+		 */
+		$json_decoded = self::command_to_array( WP_CLI::get_root_command() );
+
+		/**
+		 * If command is input as a string, then explode it into array.
+		 */
+		$command = ( 1 === count( $_ ) ) ? explode( ' ', $_[0] ) : $_;
+
+		/**
+		 * If command doesn't start with WP, then halt script execution and exit with status 1.
+		 */
+		if ( 'wp' !== $command[0] ) {
```

Comment:
It'd be nicer to make prefixing with `wp` optional by doing `if ( 'wp' === $args[0] ) { array_shift( $args ); }`